src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -588,6 +588,35 @@ def load_from_zarr(cls, folder, recording=None):
 
         return sorting_analyzer
 
+    def set_temporary_recording(self, recording: BaseRecording):
+        """
+        Sets a temporary recording object. This function can be useful to temporarily set
+        a "cached" recording object that is not saved in the SortingAnalyzer object to speed up
+        computations. Upon reloading, the SortingAnalyzer object will try to reload the recording
+        from the original location in a lazy way.
+
+
+        Parameters
+        ----------
+        recording : BaseRecording
+            The recording object to set as temporary recording.
+
+        Raises
+        ------
+        ValueError
+            _description_
+        """
+        # check that recording is compatible
+        assert check_recording_attributes_match(
+            recording, self.rec_attributes, skip_properties=True
+        ), "Recording attributes do not match."
+        assert np.array_equal(
+            recording.get_channel_locations(), self.get_channel_locations()
+        ), "Recording channel locations do not match."
+        if self._recording is not None:
+            warnings.warn("SortingAnalyzer recording is already set. This will overwrite the current recording.")
```

Comment:
```suggestion
            warnings.warn("SortingAnalyzer recording is already set. Current recording is being overwritten.")
```

I switched from future tense to present tense to try to emphasize it is happening. Future tense gives it a vibe that one could prevent this from happening but by the warning it is too late. 

Also maybe do we want to say that if save_as it used this new recording will be written to the metadata so use caution?

Reply:
Actually reading both our warnings they sound a little frightening. We are not overwriting the recording, but we are point to a new recording. Maybe we think a bit more how to say this without being scary.

Reply:
`temporarily replaced`?

Reply:
I think that sounds better. I'm still curious what would happen if I do:

```python'
analyzer.set_temp_recording(my_new_recording)
analyzer.save_as('memory')
do stuff
analyzer.save_as('zarr', xx)
```
Will that then write the temp_recording to the file instead?

Reply:
Yes, that's a really good point. Maybe keeping a flag would be good then

Reply:
@zm711 I added a `self._tmp_recording` variable, which is not used in the `save_as` if the `self._recording` is available, but the `self.recording` property returns the `tmp_recording` over the original recording if it's set, so it's used in computations

Reply:
That sounds good to me!